### ISSUE TITLE
2.4.7 / 1.4.11 Conflict between 1.4 .11 Understanding text and G159 (author-supplied focus indicator)

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag#3682]
Under "Relationship with 2.4.7 Focus visible, the [Understanding text of 1.4.11](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast) says 
> the visual focus indicator for a component must have sufficient contrast against the adjacent background when the component is focused, except where the appearance of the component is determined by the user agent and not modified by the author.

...and "sufficient contrast" means 3:1. There is no reference to the thickness of the indiactor. 

However, the test procedure of Sufficient Technique [G195: Using an author-supplied, visible focus indicator](https://www.w3.org/WAI/WCAG21/Techniques/general/G195). has this as a last step:
> 5. If the focus indicator does not have 3:1 contrast ratio with its adjacent colors, check that it is at least 2px thick.

So that creates uncertainty whether the quite frequent cases of buttons having an outside focus ring several pixels thick but below 3:1 would conform. This is an example:
![button-focus](https://github.com/w3c/wcag/assets/9989439/6971540a-909c-4f00-bae8-50e8e6e174c8)^

The light green thick focus ring has a contrast of 1.9:1 to the light grey background. Figure 9 of the 1.4.11 Understanding text seems to support the view that a focus ring with insufficient contrast to the background would not pass. If this is consensus, the test procedure of G159 needs to be revised to take out the last point in the test procedure. I am happy to create a PR for that.

**Comment by patrickhlauke** [comment]
> If the focus indicator does not have 3:1 contrast ratio with its adjacent colors, check that it is at least 2px thick.

I'm struggling to work out why that step is even there? I mean sure, for 2.4.7 Focus Visible, in theory a single faint pixel "passes". But if the technique is meant to apply also to 1.4.11 Non-Text Contrast, there is no "escape clause" built into 1.4.11 about size trumping contrast. Is this something that may have come out of early Focus Appearance stuff?

**Comment by patrickhlauke** [comment]
> the test procedure of G159 needs to be revised

absolutely in favour of that. nuke that last point, i'd say

**Comment by mraccess77** [comment]
> > If the focus indicator does not have 3:1 contrast ratio with its adjacent colors, check that it is at least 2px thick.
> 
> I'm struggling to work out why that step is even there? I mean sure, for 2.4.7 Focus Visible, in theory a single faint pixel "passes". But if the technique is meant to apply also to 1.4.11 Non-Text Contrast, there is no "escape clause" built into 1.4.11 about size trumping contrast. Is this something that may have come out of early Focus Appearance stuff?

Seems like that came from previous versions of focus appearance which is now AAA and has changed.

**Comment by patrickhlauke** [comment]
as G195 doesn't even reference Focus Appearance, that then irrelevant either way (and yes, with a contrast below 3:1, it would then *not* be sufficient for 1.4.11 which the technique is supposed to be)

**Comment by mbgower** [comment]
@patrickhlauke The [WCAG 2.2 version of G195](https://www.w3.org/WAI/WCAG22/Techniques/general/G195) _does_ indicate it is sufficient for [Focus Appearance](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance).

The [2.1 version](https://www.w3.org/WAI/WCAG21/Techniques/general/G195) does not.

So it seems we have two different issues. I agree that for the 2.1 version, the third line should be removed. I also think others do as well, unless we are aiming to exceed the requirements of Non-text Contrast.
 But if the 2.2 version is to continue to point to 2.4.13, there are other changes needed in the test to meet the AAA requirement.

**Comment by patrickhlauke** [comment]
Thanks for spotting the difference between the 2.1 and 2.2 versions of this technique. It definitely seems that for 2.1 at least, this technique is built to exceed what is normatively required - points 2., 3., specifically go beyond anything required by  1.4.11 and 2.4.7.

But yes, point 5. is wrong both for 2.1 and 2.2.

**Comment by mbgower** [comment]
The TF is planning to update the technique, preserving G195 for WCAG 2.1 and copying it over as a new technique for WCAG 2.2.

On careful review, my opinion is that the entirety of G195 (not just its test) was intended to address Focus Appearance, as it was when it was a AA criterion in a mid-life draft of WCAG 2.2.

If you look at the wording of the pertinent part of the 1.4.11 SC (shown below) and compare that to the techique's description and its tests, it is difficult to reach any other conclusion. In other words, it's main threshold is meeting Focus Apperance, and it merely meets Non-text Contrast and Focus Visible since by meeting the AAA requirement it also meets these less demanding criteria.

User Interface Components
Visual information required to identify [user interface components](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast#dfn-user-interface-component) and [states](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast#dfn-state), except for inactive components or where the appearance of the component is determined by the user agent and not modified by the author;
